test/assignment.coffee

```diff
@@ -141,9 +141,6 @@ test "#1192: assignment starting with object literals", ->
 
 # Destructuring Assignment
 
-test "empty destructuring assignment", ->
-  {} = [] = undefined
-
```

Comment:
So the test currently fails, because it compiles to:
```js
function () {
  return ({} = void 0);
}
```
which Node doesn’t like:
```
TypeError: Cannot match against 'undefined' or 'null'.
```
Adding `= []` in there generates the other error I saw before:
```
TypeError: Cannot read property 'Symbol(Symbol.iterator)' of undefined
```
So I guess the question is, so what? There’s no way to make this test “pass,” since the reasonable JavaScript that a user would expect to get generated (`{} = [] = undefined`) also throws a runtime error. Having the compiler generate anything else feels like cheating.

So either we try to have the compiler catch this error, and the test becomes whether we do catch it; or we rewrite the test to test something else that wouldn’t be a runtime error in Node. I would lean toward the latter.